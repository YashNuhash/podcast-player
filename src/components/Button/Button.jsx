import React from 'react'
import styles from './Button.module.scss'

export const Button = ({ children, inverted, ...otherProps }) => (
  <button
    type='button'
    className={`${styles.Button} ${inverted ? styles.Inverted : ''}`}
    {...otherProps}
  >
    {children}
  </button>
)

export default Button
