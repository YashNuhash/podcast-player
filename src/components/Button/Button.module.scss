@import '../../utils';

.Button {
    text-align: center;
    padding: 1rem 2rem;
    background-color: $primary;
    color: $offset;
    border: none;
    border-radius: 100px;
    font-weight: 600;
    cursor: pointer;
}

.Inverted {
    background-color: $offset;
    color: $primary;
}
