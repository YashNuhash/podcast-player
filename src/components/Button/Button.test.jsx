import React from 'react'
import { render } from '@testing-library/react'
import { Button } from './Button'

describe('Button', () => {
    const text = 'Hello'

    test('renders default version', () => {
        const { getByText } = render(<Button>{text}</Button>)
        expect(getByText(text)).toMatchSnapshot()
    })
    test('renders inverted version', () => {
        const { getByText } = render(<Button inverted>{text}</Button>)
        expect(getByText(text)).toMatchSnapshot()
    })
})
