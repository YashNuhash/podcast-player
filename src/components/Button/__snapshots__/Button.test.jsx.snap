// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders default version 1`] = `
<button
  class="Button "
  type="button"
>
  Hello
</button>
`;

exports[`Button renders inverted version 1`] = `
<button
  class="Button Inverted"
  type="button"
>
  Hello
</button>
`;
